src/PintClient/execs.ts

```diff
@@ -0,0 +1,305 @@
+      status: exec.status as ShellProcessStatus,
+    };
+  }
+  async create(
+    projectPath: string,
+    size: ShellSize,
+    command?: string,
+    type?: ShellProcessType,
+    isSystemShell?: boolean
+  ): Promise<OpenShellDTO> {
+    // For Pint, we need to construct args from command
+    const args = command ? command.split(' ').slice(1) : [];
+    const baseCommand = command ? command.split(' ')[0] : 'bash';
+    const exec = await createExec({
+      client: this.apiClient,
+      body: {
+        args,
+        command: baseCommand,
+        interactive: type === "COMMAND" ? false : true,
+      },
+    });
+
+    if (!exec.data) {
+      console.log(exec);
+      throw new Error(exec.error.message);
+    }
+
+    console.log("Gotz shell", exec.data);
+
+    await this.open(exec.data.id, { cols: 200, rows: 80 });
+
+    return {
+      ...this.convertExecToShellDTO(exec.data),
+      buffer: [],
+    };
+  }
+  async delete(shellId: ShellId): Promise<CommandShellDTO | TerminalShellDTO | null> {
+    try {
+      // First get the exec details before deleting it
+      const exec = await getExec({
+        client: this.apiClient,
+        path: {
+          id: shellId,
+        },
+      });
+
+      if (!exec.data) {
+        return null; // Exec doesn't exist
+      }
+
+      // Convert to shell DTO before deletion
+      const shellDTO = this.convertExecToShellDTO(exec.data);
+
+      // Delete the exec
+      const deleteResponse = await deleteExec({
+        client: this.apiClient,
+        path: {
+          id: shellId,
+        },
+      });
+
+      if (deleteResponse.data) {
+        // Clean up any open shells reference
+        if (this.openShells[shellId]) {
+          this.openShells[shellId].abort();
+          delete this.openShells[shellId];
+        }
+
+        return shellDTO as CommandShellDTO | TerminalShellDTO;
+      } else {
+        return null;
+      }
+    } catch (error) {
+      console.error("Failed to delete shell:", error);
+      return null;
+    }
+  }
+  async getShells(): Promise<ShellDTO[]> {
+    const execs = await listExecs({
+      client: this.apiClient,
+    });
+
+    return (
+      execs.data?.execs.map((exec) => this.convertExecToShellDTO(exec)) ?? []
+    );
+  }
+  async open(shellId: ShellId, size: ShellSize): Promise<OpenShellDTO> {
+    const abortController = new AbortController();
+
+    this.openShells[shellId] = abortController;
+
+    const exec = await getExec({
+      client: this.apiClient,
+      path: {
+        id: shellId,
+      },
+    });
+
+    if (!exec.data) {
+      throw new Error(exec.error.message);
+    }
+
+    const { stream } = await getExecOutput({
+      client: this.apiClient,
+      path: { id: shellId },
+      query: { lastSequence: 0 },
+      signal: abortController.signal,
+      headers: {
+        Accept: "text/event-stream",
+      },
+    });
+
+    const buffer: string[] = [];
+
+    console.log("Waiting for IO");
```

Comment:
You might want to remove all console logs and console errors as this can interfere with the logs of the app that uses the ask